File: backend/internal/appdto/setting.go
package appdto

type Setting struct {
	Group     string `json:"group"`
	Key       string `json:"key"`
	Value     string `json:"value"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
}

type CreateSettingReq struct {
	Group string `json:"group" binding:"required"`
	Key   string `json:"key" binding:"required"`
	Value string `json:"value" binding:"required"`
}

type DeleteSettingReq struct {
	Group string `json:"group" validate:"required,min=1,max=64"`
	Key   string `json:"key" validate:"required,min=1,max=64"`
}

type GetSettingReq struct {
	Group string `json:"group" validate:"required,min=1,max=64"`
	Key   string `json:"key" validate:"required,min=1,max=64"`
}

type GetSettingsReq struct {
	Group string `json:"group" validate:"omitempty,min=1,max=64"`
}

type LLMConfig struct {
	Provider string         `yaml:"provider" json:"provider"`
	OpenAI   OpenAIConfig   `yaml:"openai" json:"openai"`
	DeepSeek DeepSeekConfig `yaml:"deepseek" json:"deepseek"`
	Volce    VolceConfig    `yaml:"volce" json:"volce"`
}

type OpenAIConfig struct {
	APIKey  string `yaml:"api_key" json:"api_key"`
	BaseURL string `yaml:"base_url" json:"base_url"`
	Model   string `yaml:"model" json:"model"`
	OrgID   string `yaml:"org_id" json:"org_id"`
	APIType string `yaml:"api_type" json:"api_type"`
}

type DeepSeekConfig struct {
	APIKey  string `yaml:"api_key" json:"api_key"`
	BaseURL string `yaml:"base_url" json:"base_url"`
	Model   string `yaml:"model" json:"model"`
}

type VolceConfig struct {
	APIKey  string `yaml:"api_key" json:"api_key"`
	BaseURL string `yaml:"base_url" json:"base_url"`
	Model   string `yaml:"model" json:"model"`
}

type LLMSetting struct {
	*LLMConfig
}

type UpdateLLMSettingReq struct {
	*LLMConfig
}

type ChatLLMConfig struct {
	OpenAI   ChatOpenAIConfig   `yaml:"openai" json:"openai"`
	DeepSeek ChatDeepSeekConfig `yaml:"deepseek" json:"deepseek"`
	Volce    ChatVolceConfig    `yaml:"volce" json:"volce"`
}

type ChatOpenAIConfig struct {
	APIKey  string   `yaml:"api_key" json:"api_key"`
	BaseURL string   `yaml:"base_url" json:"base_url"`
	Models  []string `yaml:"models" json:"models"`
	OrgID   string   `yaml:"org_id" json:"org_id"`
	APIType string   `yaml:"api_type" json:"api_type"`
}

type ChatDeepSeekConfig struct {
	APIKey  string   `yaml:"api_key" json:"api_key"`
	BaseURL string   `yaml:"base_url" json:"base_url"`
	Models  []string `yaml:"models" json:"models"`
}

type ChatVolceConfig struct {
	APIKey  string   `yaml:"api_key" json:"api_key"`
	BaseURL string   `yaml:"base_url" json:"base_url"`
	Models  []string `yaml:"models" json:"models"`
}

type ChatLLMSetting struct {
	*ChatLLMConfig
}

type UpdateChatLLMSettingReq struct {
	*ChatLLMConfig
}

type AgentConfig struct {
	Name   string   `json:"name"`
	Prompt string   `json:"prompt"`
	Tools  []string `json:"tools"`
}

type AgentSetting struct {
	*AgentConfig
}

type UpdateAgentSettingReq struct {
	*AgentConfig
}

type SimpleMemoryConfig struct {
	MaxHistoryMessages int `json:"max_history_messages" yaml:"max_history_messages"`
}

type MongoDBMemoryConfig struct {
	URI                string `json:"uri" yaml:"uri"`
	Database           string `json:"database" yaml:"database"`
	Collection         string `json:"collection" yaml:"collection"`
	MaxHistoryMessages int    `json:"max_history_messages" yaml:"max_history_messages"`
}

type RedisMemoryConfig struct {
	Host               string `json:"host" yaml:"host"`
	Port               int    `json:"port" yaml:"port"`
	Username           string `json:"username" yaml:"username"`
	Password           string `json:"password" yaml:"password"`
	DB                 int    `json:"db" yaml:"db"`
	KeyPrefix          string `json:"key_prefix" yaml:"key_prefix"`
	MaxHistoryMessages int    `json:"max_history_messages" yaml:"max_history_messages"`
}

type SQLiteMemoryConfig struct {
	DSN                string `json:"dsn" yaml:"dsn"`
	TableName          string `json:"table_name" yaml:"table_name"`
	MaxHistoryMessages int    `json:"max_history_messages" yaml:"max_history_messages"`
}

type MySQLMemoryConfig struct {
	Host               string `json:"host" yaml:"host"`
	Port               int    `json:"port" yaml:"port"`
	Username           string `json:"username" yaml:"username"`
	Password           string `json:"password" yaml:"password"`
	Database           string `json:"database" yaml:"database"`
	TableName          string `json:"table_name" yaml:"table_name"`
	MaxHistoryMessages int    `json:"max_history_messages" yaml:"max_history_messages"`
}

type MemoryConfig struct {
	Provider string              `json:"provider" yaml:"provider"` // simple, mongodb, redis, sqlite, mysql
	Simple   SimpleMemoryConfig  `json:"simple" yaml:"simple"`
	MongoDB  MongoDBMemoryConfig `json:"mongodb" yaml:"mongodb"`
	Redis    RedisMemoryConfig   `json:"redis" yaml:"redis"`
	SQLite   SQLiteMemoryConfig  `json:"sqlite" yaml:"sqlite"`
	MySQL    MySQLMemoryConfig   `json:"mysql" yaml:"mysql"`
}

type MemorySetting struct {
	*MemoryConfig
}

type UpdateMemorySettingReq struct {
	*MemoryConfig
}

type UpdateSettingReq struct {
	Group string `json:"group" binding:"required"`
	Key   string `json:"key" binding:"required"`
	Value string `json:"value"`
}

File: frontend/src/apis/setting.ts
import { request } from "@/utils";

export interface CreateSettingRequest {
  group: string;
  key: string;
  value: string;
}

export interface UpdateSettingRequest {
  group: string;
  key: string;
  value: string;
}

export interface DeleteSettingRequest {
  group: string;
  key: string;
}

export interface GetSettingRequest {
  group: string;
  key: string;
}

export interface Setting {
  group: string;
  key: string;
  value: string;
  created_at: string;
  updated_at: string;
}

export const createSetting = (data: CreateSettingRequest) => request.post('/settings', data);

export const updateSetting = (data: UpdateSettingRequest) => request.put('/settings', data);

export const deleteSetting = (data: DeleteSettingRequest) => request.delete(`/settings/${data.group}/${data.key}`);

export const getSetting = (data: GetSettingRequest) => request.get<Setting>(`/settings/${data.group}/${data.key}`);

export const getSettings = (params?: { group?: string }) => request.get<Setting[]>('/settings', { params });

export interface LLMConfig {
  provider: string;
  openai: {
    api_key: string;
    base_url: string;
    model: string;
    org_id: string;
    api_type: string;
  };
  deepseek: {
    api_key: string;
    base_url: string;
    model: string;
  };
  volce: {
    api_key: string;
    base_url: string;
    model: string;
  };
}

export interface LLMSetting {
  provider: string;
  openai: {
    api_key: string;
    base_url: string;
    model: string;
    org_id: string;
    api_type: string;
  };
  deepseek: {
    api_key: string;
    base_url: string;
    model: string;
  };
  volce: {
    api_key: string;
    base_url: string;
    model: string;
  };
}

export interface UpdateLLMSettingRequest {
  provider: string;
  openai: {
    api_key: string;
    base_url: string;
    model: string;
    org_id: string;
    api_type: string;
  };
  deepseek: {
    api_key: string;
    base_url: string;
    model: string;
  };
  volce: {
    api_key: string;
    base_url: string;
    model: string;
  };
}

export const getLLMSetting = () => request.get<LLMSetting>('/settings/llm');

export const updateLLMSetting = (data: UpdateLLMSettingRequest) => request.put('/settings/llm', data);

export interface ChatLLMConfig {
  openai: {
    api_key: string;
    base_url: string;
    models: string[];
    org_id: string;
    api_type: string;
  };
  deepseek: {
    api_key: string;
    base_url: string;
    models: string[];
  };
  volce: {
    api_key: string;
    base_url: string;
    models: string[];
  };
}

export interface ChatLLMSetting {
  openai: {
    api_key: string;
    base_url: string;
    models: string[];
    org_id: string;
    api_type: string;
  };
  deepseek: {
    api_key: string;
    base_url: string;
    models: string[];
  };
  volce: {
    api_key: string;
    base_url: string;
    models: string[];
  };
}

export interface UpdateChatLLMSettingRequest {
  openai: {
    api_key: string;
    base_url: string;
    models: string[];
    org_id: string;
    api_type: string;
  };
  deepseek: {
    api_key: string;
    base_url: string;
    models: string[];
  };
  volce: {
    api_key: string;
    base_url: string;
    models: string[];
  };
}

export const getChatLLMSetting = (params?: { mask_sensitive?: boolean }) => request.get<ChatLLMSetting>('/settings/chat-llm', { params });

export const updateChatLLMSetting = (data: UpdateChatLLMSettingRequest) => request.put('/settings/chat-llm', data);

export interface FetchLLMModelsRequest {
  provider: string;
  api_key: string;
  base_url?: string;
}
export interface FetchLLMModelsResponse {
  models: string[];
}
export const fetchLLMModels = (data: FetchLLMModelsRequest) => request.post<FetchLLMModelsResponse>('/llm/models/fetch', data, { timeout: 120000 });

export interface AgentConfig {
  name: string;
  prompt: string;
  tools: string[];
}

export interface AgentSetting {
  name: string;
  prompt: string;
  tools: string[];
}

export interface UpdateAgentSettingRequest {
  name: string;
  prompt: string;
  tools: string[];
}

export const getAgentSetting = () => request.get<AgentSetting>('/settings/agent');

export const updateAgentSetting = (data: UpdateAgentSettingRequest) => request.put('/settings/agent', data);

export interface SimpleMemoryConfig {
  max_history_messages: number;
}

export interface MongoDBMemoryConfig {
  uri: string;
  database: string;
  collection: string;
  max_history_messages: number;
}

export interface RedisMemoryConfig {
  host: string;
  port: number;
  username?: string;
  password?: string;
  db: number;
  key_prefix: string;
  max_history_messages: number;
}

export interface SQLiteMemoryConfig {
  dsn: string;
  table_name: string;
  max_history_messages: number;
}

export interface MySQLMemoryConfig {
  host: string;
  port: number;
  username: string;
  password: string;
  database: string;
  table_name: string;
  max_history_messages: number;
}

export interface MemoryConfig {
  provider: string; // simple, mongodb, redis, sqlite, mysql
  simple: SimpleMemoryConfig;
  mongodb: MongoDBMemoryConfig;
  redis: RedisMemoryConfig;
  sqlite: SQLiteMemoryConfig;
  mysql: MySQLMemoryConfig;
}

export interface MemorySetting {
  provider: string;
  simple: SimpleMemoryConfig;
  mongodb: MongoDBMemoryConfig;
  redis: RedisMemoryConfig;
  sqlite: SQLiteMemoryConfig;
  mysql: MySQLMemoryConfig;
}

export interface UpdateMemorySettingRequest {
  provider: string;
  simple: SimpleMemoryConfig;
  mongodb: MongoDBMemoryConfig;
  redis: RedisMemoryConfig;
  sqlite: SQLiteMemoryConfig;
  mysql: MySQLMemoryConfig;
}

export const getMemorySetting = () => request.get<MemorySetting>('/settings/memory');

export const updateMemorySetting = (data: UpdateMemorySettingRequest) => request.put('/settings/memory', data);
